### ISSUE TITLE
GraphQL update auth failure

### ISSUE THREAD

**Issue by dmost714** [aws-amplify/amplify-category-api#603]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v18.3.0

### Amplify CLI Version

9.0.0

### What operating system are you using?

Mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

Yes, but I modified lambdas and this is a react front-end and/or API graphql issue.

### Amplify Categories

auth, api

### Amplify Commands

push

### Describe the bug

I have a schema that uses GROUP to determine which users have access to records.
As of amplify cli v7.6.22, updating a record fails with error "Not authorized to access updateCampaign on type Mutation"

<img width="685" alt="image" src="https://user-images.githubusercontent.com/224830/176043345-73af35bf-7105-4ec4-ab28-5d5fe2609ff4.png">

If I revert to amplify cli v7.6.21, everything works again.

v7.6.21 <-- works
v7.6.22 <-- no workie
v7.6.26 <-- no workie
v9.0.0 <-- no workie

### Expected behavior

The api call should succeed and the data should be updated.

### Reproduction steps

As I recall, you can reproduce this using GraphiQL.

Here's a snippet of the code I use to update the record.
As of v7.6.22, this code throws the error in the screenshot above.

```typescript
const update = {
  id: campaign.id,
  status: data.status,
  title: data.title,
}
try {
  const gqlData: any = await API.graphql(graphqlOperation(updateCampaign, { input: update }))
  updateCampaignData(gqlData.data.updateCampaign as Campaign)
} catch (error) {
  console.log('error', error)
  alert('update failed')
}

```

If I deploy any version of Amplify CLI later than v7.6.21 and bump a comment in the schema so that the VTLs are regenerated, the update will fail. (The VTLs are Amplify generated, I have not added/modified them.)

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line
type Campaign @model @auth(rules:[
  { allow: groups, groupsField: "group", operations: [create, read, update] },
  { allow: private, provider: iam }
])
{
  id: ID!
  group: String!
  account: Account @belongsTo
  orders: [Order] @hasMany
  accountCampaignsId: String! @index(name: "campaignsByAccountIdAndDate", queryField: "campaignsByAccountIdAndDate", sortKeyFields: ["createdAt"])
  status: CampaignStatus
  title: String!
  createdAt: AWSDateTime!
}

```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by dmost714** [comment]
Here is  **Mutation.updateOrder.req.vtl** generated from AmplifyCLI **v7.6.21**:

```
## [Start] Mutation Update resolver. **
#set( $args = $util.defaultIfNull($ctx.stash.transformedArgs, $ctx.args) )
## Set the default values to put request **
#set( $mergedValues = $util.defaultIfNull($ctx.stash.defaultValues, {}) )
## copy the values from input **
$util.qr($mergedValues.putAll($util.defaultIfNull($args.input, {})))
## set the typename **
## Initialize the vars for creating ddb expression **
#set( $expNames = {} )
#set( $expValues = {} )
#set( $expSet = {} )
#set( $expAdd = {} )
#set( $expRemove = [] )
#if( $ctx.stash.metadata.modelObjectKey )
  #set( $Key = $ctx.stash.metadata.modelObjectKey )
#else
  #set( $Key = {
  "id":   $util.dynamodb.toDynamoDB($args.input.id)
} )
#end
## Model key **
#if( $ctx.stash.metadata.modelObjectKey )
  #set( $keyFields = [] )
  #foreach( $entry in $ctx.stash.metadata.modelObjectKey.entrySet() )
    $util.qr($keyFields.add("$entry.key"))
  #end
#else
  #set( $keyFields = ["id"] )
#end
#foreach( $entry in $util.map.copyAndRemoveAllKeys($mergedValues, $keyFields).entrySet() )
  #if( !$util.isNull($ctx.stash.metadata.dynamodbNameOverrideMap) && $ctx.stash.metadata.dynamodbNameOverrideMap.containsKey("$entry.key") )
    #set( $entryKeyAttributeName = $ctx.stash.metadata.dynamodbNameOverrideMap.get("$entry.key") )
  #else
    #set( $entryKeyAttributeName = $entry.key )
  #end
  #if( $util.isNull($entry.value) )
    #set( $discard = $expRemove.add("#$entryKeyAttributeName") )
    $util.qr($expNames.put("#$entryKeyAttributeName", "$entry.key"))
  #else
    $util.qr($expSet.put("#$entryKeyAttributeName", ":$entryKeyAttributeName"))
    $util.qr($expNames.put("#$entryKeyAttributeName", "$entry.key"))
    $util.qr($expValues.put(":$entryKeyAttributeName", $util.dynamodb.toDynamoDB($entry.value)))
  #end
#end
#set( $expression = "" )
#if( !$expSet.isEmpty() )
  #set( $expression = "SET" )
  #foreach( $entry in $expSet.entrySet() )
    #set( $expression = "$expression $entry.key = $entry.value" )
    #if( $foreach.hasNext() )
      #set( $expression = "$expression," )
    #end
  #end
#end
#if( !$expAdd.isEmpty() )
  #set( $expression = "$expression ADD" )
  #foreach( $entry in $expAdd.entrySet() )
    #set( $expression = "$expression $entry.key $entry.value" )
    #if( $foreach.hasNext() )
      #set( $expression = "$expression," )
    #end
  #end
#end
#if( !$expRemove.isEmpty() )
  #set( $expression = "$expression REMOVE" )
  #foreach( $entry in $expRemove )
    #set( $expression = "$expression $entry" )
    #if( $foreach.hasNext() )
      #set( $expression = "$expression," )
    #end
  #end
#end
#set( $update = {} )
$util.qr($update.put("expression", "$expression"))
#if( !$expNames.isEmpty() )
  $util.qr($update.put("expressionNames", $expNames))
#end
#if( !$expValues.isEmpty() )
  $util.qr($update.put("expressionValues", $expValues))
#end
## Begin - key condition **
#if( $ctx.stash.metadata.modelObjectKey )
  #set( $keyConditionExpr = {} )
  #set( $keyConditionExprNames = {} )
  #foreach( $entry in $ctx.stash.metadata.modelObjectKey.entrySet() )
    $util.qr($keyConditionExpr.put("keyCondition$velocityCount", {
  "attributeExists": true
}))
    $util.qr($keyConditionExprNames.put("#keyCondition$velocityCount", "$entry.key"))
  #end
  $util.qr($ctx.stash.conditions.add($keyConditionExpr))
#else
  $util.qr($ctx.stash.conditions.add({
  "id": {
      "attributeExists": true
  }
}))
#end
## End - key condition **
#if( $args.condition )
  $util.qr($ctx.stash.conditions.add($args.condition))
#end
## Start condition block **
#if( $ctx.stash.conditions && $ctx.stash.conditions.size() != 0 )
  #set( $mergedConditions = {
  "and": $ctx.stash.conditions
} )
  #set( $Conditions = $util.parseJson($util.transform.toDynamoDBConditionExpression($mergedConditions)) )
  #if( $Conditions.expressionValues && $Conditions.expressionValues.size() == 0 )
    #set( $Conditions = {
  "expression": $Conditions.expression,
  "expressionNames": $Conditions.expressionNames
} )
  #end
  ## End condition block **
#end
#set( $UpdateItem = {
  "version": "2018-05-29",
  "operation": "UpdateItem",
  "key": $Key,
  "update": $update
} )
#if( $Conditions )
  #if( $keyConditionExprNames )
    $util.qr($Conditions.expressionNames.putAll($keyConditionExprNames))
  #end
  $util.qr($UpdateItem.put("condition", $Conditions))
#end
$util.toJson($UpdateItem)
## [End] Mutation Update resolver. **

```

**Comment by dmost714** [comment]
File **Mutation.updateOrder.auth.1.req.vtl** from v7.6.20 appears to have an ELSE condition that gets removed in v7.6.22.

```
#if( $util.authType() == "User Pool Authorization" )
  #if( !$isAuthorized )
    #set( $groupEntity0 = $util.defaultIfNull($ctx.result.group, null) )
    #set( $groupClaim0 = $util.defaultIfNull($ctx.identity.claims.get("cognito:groups"), []) )
    #set( $groupAllowedFields0 = [] )
    #set( $groupNullAllowedFields0 = [] )
    #if( $util.isString($groupClaim0) )
      #if( $util.isList($util.parseJson($groupClaim0)) )
        #set( $groupClaim0 = $util.parseJson($groupClaim0) )
      #else
        #set( $groupClaim0 = [$groupClaim0] )
      #end
    #end
    #foreach( $userGroup in $groupClaim0 )
      #if( $groupEntity0 == $userGroup )
        #if( !$groupAllowedFields0.isEmpty() || !$groupNullAllowedFields0.isEmpty() )
          $util.qr($allowedFields.addAll($groupAllowedFields0))
          $util.qr($nullAllowedFields.addAll($groupNullAllowedFields0))
        #else
          #set( $isAuthorized = true )
          #break
        #end
      #end
    #end
  #end
#end

```
<img width="453" alt="image" src="https://user-images.githubusercontent.com/224830/176060616-f5c24fce-1b77-4ce0-b796-07cf9d691a1a.png">

**Comment by dmost714** [comment]
@josefaidt Ack. I believe this is in fact due to inclusion of NULL fields in the update. I'm not sure why it was working in older versions, but after updating my code to prevent null keys from being included in the update; things are working in v9.1.0.

While I suppose it is true that my code is not "authorized" to send up nulls on those fields, the "unauthorized" error message is extremely misleading.

I'm closing this issue.
